controllers/release/adapter_test.go

```diff
@@ -656,58 +432,150 @@ var _ = Describe("Release Adapter", Ordered, func() {
+			Expect(adapter.release.IsReleasing()).To(BeTrue())
 		})
 
-		It("has owner annotations", func() {
-			Expect(pipelineRun.GetAnnotations()[handler.NamespacedNameAnnotation]).To(ContainSubstring(adapter.release.Name))
-			Expect(pipelineRun.GetAnnotations()[handler.TypeAnnotation]).To(ContainSubstring("Release"))
-		})
+		It("should do nothing if the release is already running", func() {
+			adapter.release.MarkReleasing("")
 
-		It("has release labels", func() {
-			Expect(pipelineRun.GetLabels()[tekton.PipelinesTypeLabel]).To(Equal("release"))
-			Expect(pipelineRun.GetLabels()[tekton.ReleaseNameLabel]).To(Equal(adapter.release.Name))
-			Expect(pipelineRun.GetLabels()[tekton.ReleaseNamespaceLabel]).To(Equal(testNamespace))
+			result, err := adapter.EnsureReleaseIsRunning()
+			Expect(!result.RequeueRequest && !result.CancelRequest).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
+			Expect(adapter.release.IsReleasing()).To(BeTrue())
 		})
+	})
 
-		It("references the pipeline specified in the ReleaseStrategy", func() {
-			Expect(pipelineRun.Spec.PipelineRef.Name).To(Equal(releaseStrategy.Spec.Pipeline))
+	Context("When EnsureReleaseIsProcessed is called", func() {
+		var adapter *Adapter
+
+		AfterEach(func() {
+			_ = adapter.client.Delete(ctx, adapter.release)
 		})
 
-		It("contains a parameter with the json representation of the EnterpriseContractPolicy", func() {
-			jsonSpec, _ := json.Marshal(enterpriseContractPolicy.Spec)
-			Expect(pipelineRun.Spec.Params).Should(ContainElement(HaveField("Value.StringVal", Equal(string(jsonSpec)))))
+		BeforeEach(func() {
+			adapter = createReleaseAndAdapter()
 		})
 
-		It("contains a parameter with the json representation of the Snapshot", func() {
-			jsonSpec, _ := json.Marshal(snapshot.Spec)
-			Expect(pipelineRun.Spec.Params).Should(ContainElement(HaveField("Value.StringVal", Equal(string(jsonSpec)))))
+		It("should do nothing if the Release is already processed", func() {
+			adapter.release.MarkProcessing("")
+			adapter.release.MarkProcessed()
+
+			result, err := adapter.EnsureReleaseIsRunning()
+			Expect(!result.RequeueRequest && !result.CancelRequest).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
+			Expect(adapter.release.IsProcessing()).To(BeFalse())
+		})
+
+		It("should continue if the PipelineRun exists and the release processing has started", func() {
+			adapter.ctx = loader.GetMockedContext(ctx, []loader.MockData{
+				{
+					ContextKey: loader.ReleasePipelineRunContextKey,
+					Resource: &v1beta1.PipelineRun{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "pipeline-run",
+							Namespace: "default",
+						},
+					},
+				},
+			})
+			adapter.release.MarkProcessing("")
+
+			result, err := adapter.EnsureReleaseIsProcessed()
+			Expect(!result.RequeueRequest && !result.CancelRequest).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should register the processing data if the pipelineRun already exists", func() {
```

Comment:
on line 493 you capitalize it as PipelineRun. Let's be consistent

Reply:
Agree.